lms/djangoapps/program_enrollments/signals.py

```diff
@@ -0,0 +1,57 @@
+"""
+Signal handlers for program enrollments
+"""
+import logging
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+from social_django.models import UserSocialAuth
+from student.models import CourseEnrollmentException
+from third_party_auth.models import SAMLProviderConfig
+
+from lms.djangoapps.program_enrollments.models import ProgramEnrollment
+
+logger = logging.getLogger(__name__)
+
+
+@receiver(post_save, sender=UserSocialAuth)
+def matriculate_learner(sender, instance, **kwargs):  # pylint: disable=unused-argument
+    """
+    Post-save signal to update any waiting program enrollments with a user,
+    and enroll the user in any waiting course enrollments.
+
+    In most cases this will just short-circuit. Enrollments will only be updated
+    for a SAML provider with a linked organization.
+    """
+    try:
+        user = instance.user
+        provider_slug, external_user_key = instance.uid.split(':')
+        if not SAMLProviderConfig.objects.with_active_flag().filter(is_active=1).get(slug=provider_slug).organization:
+            return
+    except (AttributeError, ValueError, SAMLProviderConfig.DoesNotExist):
+        return
```

Comment:
Could you please log an error here to help debugging?

Reply:
so this isn't really an error.  This is just the short circuit that will happen for every social auth entry not related to masters.  Seems like it could get noisy if I log here?

Reply:
Oh, I see, you also have `SAMLProviderConfig.DoesNotExist` exception case here. That said, I don't believe log here will get noisy because, how many times do social auth link gets created? like 2k in a month maybe? That's nothing in terms of logs.

Reply:
taking a second look at this and I'm thinking we should log if we get an incoming social auth that looks like saml, but no configuration exists (`SAMLProviderConfig.DoesNotExist`).  That sounds like an exceptional case we might want to know about.  If the uid format is something else like google (AttributeError/ValueError) we can probably just drop that one b/c it's not related to this feature at all.  Does that make sense to you?

Reply:
I see. Make sense.